delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/stream/ElemErrorStore.scala

```diff
@@ -0,0 +1,138 @@
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.Elem.FailedElem
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.ElemErrorStore.ElemErrorRow
+import ch.epfl.bluebrain.nexus.delta.sourcing.implicits.IriInstances._
+import doobie._
+import doobie.implicits._
+import doobie.postgres.implicits._
+import fs2.Stream
+import monix.bio.Task
+
+import java.time.Instant
+
+trait ElemErrorStore {
+
+  /**
+    * Save one error
+    */
+  def save(metadata: ProjectionMetadata, failure: FailedElem): ConnectionIO[Unit]
+
+  /**
+    * Get all errors for a given projection id
+    */
+  def entries(
+      projectionProject: Option[ProjectRef],
+      projectionId: String,
+      offset: Offset
+  ): Stream[Task, ElemErrorRow]
+
+  /**
+    * Get all errors start from the given offset
+    */
+  def entries(projectionName: String, offset: Offset): Stream[Task, ElemErrorRow]
+
+}
+
+object ElemErrorStore {
+
+  def apply(xas: Transactors, config: QueryConfig): ElemErrorStore =
+    new ElemErrorStore {
+
+      override def entries(
+          projectionProject: Option[ProjectRef],
+          projectionId: String,
+          offset: Offset
+      ): Stream[Task, ElemErrorRow] =
+        sql"""SELECT * from public.elem_errors
+             |WHERE projection_id = $projectionId
+             |AND ordering >= ${offset.value}
+             |ORDER BY ordering DESC""".stripMargin
+          .query[ElemErrorRow]
+          .streamWithChunkSize(config.batchSize)
+          .transact(xas.streaming)
+
+      override def entries(
+          projectionName: String,
+          offset: Offset
+      ): Stream[Task, ElemErrorRow] =
+        sql"""SELECT * from public.elem_errors
+             |WHERE projection_name = $projectionName
+             |AND ordering >= ${offset.value}
+             |ORDER BY ordering DESC""".stripMargin
+          .query[ElemErrorRow]
+          .streamWithChunkSize(config.batchSize)
+          .transact(xas.streaming)
+
+      override def save(
+          metadata: ProjectionMetadata,
+          failure: FailedElem
+      ): ConnectionIO[Unit] =
+        sql"""
+               | INSERT INTO public.elem_errors (
+               |  projection_name,
+               |  projection_module,
+               |  projection_project,
+               |  projection_id,
+               |  entity_type,
+               |  elem_offset,
+               |  elem_id,
+               |  message,
+               |  stack_trace,
+               |  instant
+               | )
+               | VALUES (
+               |  ${metadata.name},
+               |  ${metadata.module},
+               |  ${metadata.project},
+               |  ${metadata.resourceId},
+               |  ${failure.tpe.value},
+               |  ${failure.offset.value},
+               |  ${failure.id},
+               |  ${failure.throwable.getMessage},
+               |  ${failure.throwable.getStackTrace.map(_.toString).toString},
```

Comment:
How does the stacktrace look like like this ?
We used something like this before:
```scala
def stackTraceAsString(t: Throwable): String = {
    val sw = new StringWriter
    t.printStackTrace(new PrintWriter(sw))
    sw.toString
  }
```